lib/authentication_methods.rb

```diff
@@ -246,6 +245,31 @@ def require_user
   end
   protected :require_user
 
+  def require_user_render_json_unauthorized
+    if @current_user && @current_pseudonym
+      true
+    else
+      render :json => {
+               :status => I18n.t('lib.auth.status_unauthenticated', 'unauthenticated'),
+               :errors => [ { :message => I18n.t('lib.auth.authentication_required', "user authentication required") } ],
+               :login_url => effective_cas_login_url 
+             },
+             :status => :unauthorized
+    end
+  end
+  protected :require_user_render_json_unauthorized
+
+  def effective_cas_login_url
+    #... so if SSO is set, we can send them there directly, instead of through
+    # two other Canvas middleman pages, shaving a noticiable amount of time off
+    # for the user to experience
+    if @domain_root_account.auth_discovery_url
```

Comment:
Note that I aspire to do a followup PR where if there is an expired session, I figure out which login controller to send them to instead of sending them to the discovery page. Everyone should go to `login/cas/1` except NLU students. It would be such a nicer UX! But this will prob get dereprioritized considering I fixed the "user gets logged out after 24 hours" bug, so it won't be happening all the time like it used to.

Reply:
If I remember correct, I believe you can handle this using the service ticket looking for `EmployeeNumber`.